packages/components/src/Button/Button.tsx

```diff
@@ -19,75 +19,73 @@ interface ButtonFoundationProps {
   readonly ariaHaspopup?: boolean;
   readonly ariaExpanded?: boolean;
   readonly disabled?: boolean;
-  readonly external?: boolean;
   readonly fullWidth?: boolean;
   readonly icon?: IconNames;
   readonly iconOnRight?: boolean;
   readonly id?: string;
   readonly label?: string;
   readonly loading?: boolean;
   readonly size?: "small" | "base" | "large";
-  onClick?(
-    event: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement>,
-  ): void;
   onMouseDown?(
     event: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement>,
   ): void;
 }
 
-interface ButtonIconProps extends ButtonFoundationProps {
+interface ButtonIconProps {
   readonly icon: IconNames;
   readonly ariaLabel: string;
 }
-
 interface ButtonLabelProps extends ButtonFoundationProps {
   readonly label: string;
 }
 
-interface ButtonAnchorProps extends ButtonFoundationProps {
+interface BaseActionProps {
+  readonly variation?: "work" | "learning" | "subtle" | "destructive";
+  readonly type?: "primary" | "secondary" | "tertiary";
+  onClick?(
+    event: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement>,
+  ): void;
+}
+
+interface ButtonAnchorProps extends BaseActionProps {
   /**
    * Used to create an 'href' on an anchor tag.
    */
   readonly url?: string;
+  /**
+   * Used to open a new tab instead of change the location of the current tab
+   */
+  readonly external?: boolean;
 }
 
-interface ButtonLinkProps extends ButtonFoundationProps {
+interface ButtonLinkProps extends BaseActionProps {
   /**
    * Used for client side routing. Only use when inside a routed component.
    */
   readonly to?: string;
 }
 
-interface BaseActionProps extends ButtonFoundationProps {
-  readonly variation?: "work" | "learning" | "subtle" | "destructive";
-  readonly type?: "primary" | "secondary" | "tertiary";
-}
-
-interface DestructiveActionProps extends ButtonFoundationProps {
-  readonly variation: "destructive";
-  readonly type?: "primary" | "secondary" | "tertiary";
-}
-
-interface SubmitActionProps
-  extends Omit<ButtonFoundationProps, "external" | "onClick"> {
-  readonly type?: "primary";
-  readonly submit: boolean;
-}
-
-interface SubmitButtonProps
-  extends Omit<ButtonFoundationProps, "external" | "onClick"> {
+interface SubmitButtonProps {
   /**
    * Allows the button to submit a form
    */
-  submit: boolean;
+  submit: true;
+  readonly type?: "primary";
+  readonly variation?: "work";
```

Comment:
It was not obvious before that submit buttons must have the variation "work" however variation could not be added and the component defaults variation to "work" I made this change so `type` and `variation` are consistent.

I have decided not to change existing functionality, though it was discussed [here](https://getjobber.slack.com/archives/C03BJHV3PMG/p1679074908755019?thread_ts=1679073619.972249&cid=C03BJHV3PMG) that we might want to be less restrictive in the future. At which point we would define `type` and `variation` on the ButtonFoundationProps